File: l001_Basics/l002Basics.java
import java.util.Scanner;

public class l002Basics {

  public static Scanner scn = new Scanner(System.in);

  // Q1)
  public static void printZ() {
    System.out.println("*****");
    System.out.println("   *");
    System.out.println("  *");
    System.out.println(" *");
    System.out.println("*****");
  }

  // Q2)
  public static void gradingSystem(int marks) {
    if (marks > 90) {
      System.out.println("excellent");
    } else if (marks > 80 && marks <= 90) {
      System.out.println("good");
    } else if (marks > 70 && marks <= 80) {
      System.out.println("fair");
    } else if (marks > 60 && marks <= 70) {
      System.out.println("meets expectations");
    } else {
      System.out.println("below par");
    }
  }

  // Class 2 -> loops
  public static void printNTimes(int n) {
    for (int i = 1; i <= n; i++) {
      System.out.println("Hello");
    }
    System.out.println();
  }

  // Q2)
  public static void printTableofN(int n) {
    for (int i = 1; i <= 10; i++) {
      // cout << n << " X " << i << " = " << n * i << endl;

      System.out.println((n) + " X " + (i) + " = " + (n * i));
      // try to use this 2nd way because hame generally string return krne ke liye bolte hai
    }
    System.out.println();
  }

  // Q3)
  public static void printTableNTimes(int n) {
    for (int i = 1; i <= n; i++) {
      printTableofN(i);
    }
    System.out.println();
  }

  // Q4)
  public static void oddEven(int n) {
    if (n % 2 == 0) {
      System.out.println("even");
    } else {
      System.out.println("odd");
    }
  }

  // Q5)
  public static boolean isPrime(int n) {
    for (int i = 2; i * i <= n; i++) {
      if (n % i == 0) return false;
    }
    /*
     * 2 is only even prime number
     * prime number are natural numbers between 2 to n - 1; either it is divisble by 1  or number itself
     *
     */
    return true;
  }

  public static void primeNumber() {
    int t = scn.nextInt();
    for (int i = 1; i <= t; i++) {
      int n = scn.nextInt();
      if (isPrime(n)) {
        System.out.println(n + " - " + "Number is prime");
      } else {
        System.out.println(n + " - " + "Number is Not prime");
      }
    }
  }

  // Q6
  public static void printPrimeTill(int low, int high) {
    for (int i = low; i <= high; i++) {
      if (isPrime(i)) {
        System.out.print(i + " ");
      }
    }
  }

  // Q7
  public static void fibonacciSeries(int n) {
    int a = 0, b = 1;
    for (int i = 1; i <= n; i++) {
      System.out.print(a + " ");

      int c = a + b;
      a = b;
      b = c;
    }
  }

  // Q8
  public static int countDigits(int n) {
    
  }

  public static void main(String[] args) {
    // printZ();
    int n = scn.nextInt();
    // gradingSystem(n);
    // oddEven(n);
    // primeNumber();
    // printPrimeTill(2, 10);
    fibonacciSeries(n);
  }
}

File: l001_Basics/l002Basics.cpp
#include <iostream>
using namespace std;

// Q1)
void printZ()
{
    cout << "*****" << endl;
    cout << "   *" << endl;
    cout << "  *" << endl;
    cout << " *" << endl;
    cout << "*****" << endl;
}

// Q2)
void gradingSystem(int marks)
{
    if (marks > 90)
    {
        cout << "excellent" << endl;
    }
    else if (marks > 80 && marks <= 90)
    {
        cout << "good" << endl;
    }
    else if (marks > 70 && marks <= 80)
    {
        cout << "fair" << endl;
    }
    else if (marks > 60 && marks <= 70)
    {
        cout << "meets expectations" << endl;
    }
    else
    {
        cout << "below par" << endl;
    }
}

// Class 2 -> loops
void printNTimes(int n)
{
    for (int i = 1; i <= n; i++)
    {
        cout << "Hello" << endl;
    }
    cout << endl;
}

// Q2)
void printTableofN(int n)
{
    for (int i = 1; i <= 10; i++)
    {
        // cout << n << " X " << i << " = " << n * i << endl;

        cout << to_string(n) + " X " << to_string(i) << " = " << to_string(n * i) << endl;
        // try to use this 2nd way because hame generally string return krne ke liye bolte hai
    }
    cout << endl;
}

// Q3)
void printTableNTimes(int n)
{
    for (int i = 1; i <= n; i++)
    {
        printTableofN(i);
    }
    cout << endl;
}

// Q4)
void oddEven(int n)
{
    if (n % 2 == 0)
    {
        cout << to_string(n) + " -> Even " << endl;
    }
    else
    {
        cout << to_string(n) + " -> Odd " << endl;
    }
}

void checkNForOddEven(int n)
{
    for (int i = 1; i <= n; i++)
    {
        int a;
        cin >> a; // 15 input liya and 15 number aayege input m, 1 1 krke, and ha number k liye btana h odd h ya even
        oddEven(a);
    }
}

// Q5
bool isPrime(int n)
{
    for (int i = 2; i * i <= n; i++)
    {
        if (n % i == 0)
        // % it gives remainder value jo ik dum last me hoti h
            return false;
    }

    return true;
}

void primeNumbers()
{
    int t;
    cin >> t;

    for (int i = 1; i <= t; i++)
    {
        // t number ke liye input liye
        int n;
        cin >> n;

        if (isPrime(n))
        {
            cout << to_string(n) + "-> Prime " << endl;
        }
        else
        {
            cout << to_string(n) + "-> Not Prime " << endl;
        }
    }
}

// Q6

int main()
{
    /*
    int marks;
    cin >> marks;
    printZ();

    gradingSystem(marks);
    */

    // class 2
    // int n;
    // cin >> n;
    // printNTimes(n);
    // printTableofN(n);
    // printTableNTimes(n);
    // checkNForOddEven(n);

    primeNumbers();
}